src/types/impl-trait.md

```diff
@@ -111,10 +123,189 @@ fn foo() -> impl Trait {
 doesn't allow the caller to determine the return type.
 Instead, the function chooses the return type, but only promises that it will implement `Trait`.
 
+
+### Type Alias Impl Trait
+
+In contrast to `impl Trait`s in function return types, type aliases can be used in more places than just return types and the same type alias can be used multiple times.
+
+```rust,ignore
+fn foo() -> impl Trait {
+    value_of_type_that_implements_Trait
+}
+
+// The above function can be changed to the
+// following without a change in behaviour for callers.
+
+type Foo = impl Trait;
+fn foo() -> Foo {
+    value_of_type_that_implements_Trait
+}
+```
+
+When such a type alias is used as the return type of multiple functions, all functions must use the same hidden type.
```

Comment:
I think it would be good to start this section with something that defines "hidden type" as the real type behind a TAIT.

Reply:
hmm... RPITs have the same thing, so I could bring it up where "abstract types" are explained?